packages/emitter-framework/src/typescript/components/interface-member.tsx

```diff
@@ -8,39 +9,50 @@ import { TypeExpression } from "./type-expression.js";
 export interface InterfaceMemberProps {
   type: ModelProperty | Operation;
   optional?: boolean;
+  refkey?: ay.Refkey;
 }
 
-export function InterfaceMember({ type, optional }: InterfaceMemberProps) {
-  const namer = useTSNamePolicy();
-  const name = namer.getName(type.name, "object-member-getter");
+export function InterfaceMember(props: InterfaceMemberProps) {
+  const namer = ts.useTSNamePolicy();
+  const name = namer.getName(props.type.name, "object-member-getter");
 
-  if ($.modelProperty.is(type)) {
-    const optionality = optional === true || type.optional === true ? "?" : "";
-
-    if (isNeverType(type.type)) {
+  if ($.modelProperty.is(props.type)) {
+    if (isNeverType(props.type.type)) {
       return null;
     }
 
-    let unpackedType = type.type;
-    const part = getHttpPart($.program, type.type);
+    let unpackedType = props.type.type;
+    const part = getHttpPart($.program, props.type.type);
     if (part) {
       unpackedType = part.type;
     }
 
     return (
-      <>
-        "{name}"{optionality}: <TypeExpression type={unpackedType} />;
-      </>
+      <ts.InterfaceMember
+        name={name}
+        optional={props.optional === true || props.type.optional === true}
+        type={<TypeExpression type={unpackedType} />}
+        refkey={ay.refkey(props.type)}
+        readonly={Boolean(props.type.decorators?.find((d) => d.decorator.name === "@readonly"))}
```

Comment:
we don't have this decorator or exact concept in typespec, i think for now can keep things as not readonly